lib/samson/secrets/key_resolver.rb

```diff
@@ -11,23 +13,29 @@ def initialize(project, deploy_groups)
 
       # expands a key by finding the most specific value for it
       # bar -> production/my_project/pod100/bar
-      def expand!(secret_key)
-        key = secret_key.split('/', 2).last
+      def expand(env_name, secret_key)
+        if env_name.end_with?(WILDCARD) ^ secret_key.end_with?(WILDCARD)
+          @errors << "#{env_name} and #{secret_key} need to both end with #{WILDCARD} or not include them"
```

Comment:
2.3.1 :008 > WILDCARD='*'
 => "*" 
2.3.1 :009 > env_name="FOO_*"
 => "FOO_*" 
2.3.1 :010 > secret_key="blah*"
 => "blah*" 
2.3.1 :011 > env_name.end_with?(WILDCARD) ^ secret_key.end_with?(WILDCARD)
 => false 

does this work?